src/org/opensolaris/opengrok/web/Statistics.java

```diff
@@ -103,11 +126,11 @@ protected void maybeRefresh() {
      * @param category category
      */
     public void addRequest(HttpServletRequest req, String category) {
-        Long val = requestCategories.get(category);
+        AtomicLong val = requestCategories.get(category);
         if (val == null) {
-            val = new Long(0);
+            val = new AtomicLong(0);
```

Comment:
yep, certainly more robust. For the simple counters, the `atomic` types would work, for the maps and complex statements (like `maybeRefresh()`) probably not so much.